lib/spack/spack/hooks/dashboards.py

```diff
@@ -0,0 +1,576 @@
+        TestResult.ERRORED: 'failed',
+    }
+
+    def __init__(self, classname, name):
+        self.name = name
+        self.element = ET.Element('Test')
+        name_element = ET.SubElement(self.element, 'Name')
+        name_element.text = name
+
+        #path_element = ET.SubElement(self.element, 'Path')
+        #path_element.text = ""
+        cmd_line_element = ET.SubElement(self.element, 'FullCommandLine')
+        cmd_line_element.text = "spack install"
+        self.result = ET.SubElement(self.element, 'Results')
+
+    def set_duration(self, duration):
+        execution_time = ET.SubElement(self.result, 'NamedMeasurement')
+        execution_time.set('type', 'numeric/double')
+        execution_time.set('name', 'Execution Time')
+        value = ET.SubElement(execution_time, 'Value')
+        value.text = str(duration)
+
+    def set_result(self, result_type,
+                   message=None, error_type=None, text=None):
+        self.element.set('Status', CDashTestCase.results[result_type])
+        # Completion Status
+        completion_status = ET.SubElement(self.result, 'NamedMeasurement')
+        completion_status.set('type', 'text/string')
+        completion_status.set('name', 'Completion Status')
+        value = ET.SubElement(completion_status, 'Value')
+        value.text = 'Completed'
+        # Command line
+        cmd_line = ET.SubElement(self.result, 'NamedMeasurement')
+        cmd_line.set('type', 'text/string')
+        cmd_line.set('name', 'Command Line')
+        value = ET.SubElement(cmd_line, 'Value')
+        value.text = "spack install"
+        # Logs
+        if message is not None:
+            logs = ET.SubElement(self.result, 'Measurement')
+            value = ET.SubElement(logs, 'Value')
+            value.text = message
+
+
+
+
+class CDashSimpleTestSuite(object):
+    def __init__(self, spec, filename, slot='Experimental'):
+        self.spec = spec
+        self.slot = slot
+        self.tests = []
+        self.buildstamp = "%s-%s" % (time.strftime("%Y%d%m-%H:%M:%S"), slot)
+        self.filename = filename
+
+    def create_filename(self, spec, subdir, step):
+        if "build" in str(step):
+            if self.filename is not None:
+                return "%s.%s.xml" % (self.filename, step)
+            else:
+                fmt = '%s-{x.name}-{x.version}-{hash}.xml' % step
+                basename = fmt.format(x=spec, hash=spec.dag_hash())
+                dirname = fs.join_path(spack.var_path, subdir)
+                fs.mkdirp(dirname)
+                return fs.join_path(dirname, basename)
+
+    def create_template(self):
+        template = ET.Element('Site')
+        buildName = str(self.spec.short_spec)
+        buildName = buildName.split('=')
+        template.set('BuildName',str(buildName[0]) + " " + str(buildName[1]))
+        template.set('BuildStamp', self.buildstamp)
+        tempcompiler = self.spec.short_spec.split('%')[1].split(' ')[0]
```

Comment:
avoid using `split` and use the actual fields on .  e.g. `self.spec.compiler.name`, `self.spec.compiler.version`, etc.